src/NWCClient.ts

```diff
@@ -1084,4 +1122,59 @@ export class NWCClient {
       );
     }
   }
+
+  private async _checkCompatibility() {
+    if (!this.version) {
+      const walletServiceInfo = await this.getWalletServiceInfo();
+      const compatibleVersion = this.selectHighestCompatibleVersion(
+        walletServiceInfo.versions,
+      );
+      if (!compatibleVersion) {
+        throw new Nip47UnsupportedVersionError(
+          `no compatible version found between wallet and client`,
+          "UNSUPPORTED_VERSION",
+        );
+      }
+      this.version = compatibleVersion;
```

Comment:
if the version is 0.0 should we log a warning?

Reply:
Yeah I think it's good to add that, done!